src/Domain/Structure/ChildSize.hpp

```diff
@@ -0,0 +1,18 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "Domain/Structure/SegmentId.hpp"
+#include "NumericalAlgorithms/Spectral/Projection.hpp"
```

Comment:
please change the include of Matrix.hpp to a forward declaration in Projection.hpp